widget/completion_entry.go

```diff
@@ -0,0 +1,197 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// CompletionEntry is an Entry with options displayed in a PopUpMenu.
+type CompletionEntry struct {
+	widget.Entry
+	popupMenu     *widget.PopUp
+	navigableList *navigableList
+	Options       []string
+	pause         bool
+	itemHeight    float32
+}
+
+// NewCompletionEntry creates a new CompletionEntry which creates a popup menu that responds to keystrokes to navigate through the items without losing the editing ability of the text input.
+func NewCompletionEntry(options []string) *CompletionEntry {
+	c := &CompletionEntry{Options: options}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+// ShowCompletion displays the completion menu
+func (c *CompletionEntry) ShowCompletion() {
+	if c.pause {
+		return
+	}
+
```

Comment:
I think Completion should be hidden if there is no Options to show, otherwise you will see an empty overlay (if the popup menu was previously shown), so adding these lines maybe make sense?:
```go
if c.pause {
	return
}
if len(c.Options) == 0 {
	c.HideCompletion()
	return
}
```

Reply:
I'm not sure about this, I think that the best to do is to leave the user to choose. So it's only needed to hide the completion when the user decide it.

Reply:
Ha right...

Reply:
I did changes in  https://github.com/fyne-io/fyne-x/pull/11/commits/4056661a28e28c8488097361680b52a6e59637b8

Reply:
Great thanks :)